File: go/leap/leap.go
package leap

const testVersion = 3

func IsLeapYear(year int) bool {
	// Used: https://support.microsoft.com/en-us/help/214019/method-to-determine-whether-a-year-is-a-leap-year

	if year % 4 != 0 {
		return false
	}

	if year % 100 != 0 {
		return true
	}

	if year % 400 == 0 {
		return true
	}

	return false	

}

File: python/leap/leap.py
def is_leap_year(value: int):
    # Used: https://support.microsoft.com/en-us/help/214019/method-to-determine-whether-a-year-is-a-leap-year

    if value % 4 != 0:
        return False

    if value % 100 != 0:
        return True

    if value % 400 == 0:
        return True

    return False